msi/GitProduct.wxs

```diff
@@ -27,13 +28,64 @@
+      <RegistrySearch Id="TerminalSearch" Root="HKLM" Key="SOFTWARE\GitForWindows" Name="Terminal" Type="raw" />
+    </Property>
+    <SetProperty Id="CMDLINE_TERMINAL" Value="[TERMINAL]" Before="AppSearch" Sequence="first" />
+    <SetProperty Id="TERMINAL" Value="[CMDLINE_TERMINAL]" After="AppSearch" Sequence="first">CMDLINE_TERMINAL</SetProperty>
+
+    <!-- ENABLEFSCACHE Property implementing the "Remember Property" pattern -->
+    <Property Id="ENABLEFSCACHE" Value="1">
+      <RegistrySearch Id="EnableFsCacheSearch" Root="HKLM" Key="SOFTWARE\GitForWindows" Name="EnableFsCache" Type="raw" />
+    </Property>
+    <SetProperty Id="CMDLINE_ENABLEFSCACHE" Value="[ENABLEFSCACHE]" Before="AppSearch" Sequence="first" />
+    <SetProperty Id="ENABLEFSCACHE" Value="[CMDLINE_ENABLEFSCACHE]" After="AppSearch" Sequence="first">CMDLINE_ENABLEFSCACHE</SetProperty>
+
+    <!-- ENABLEGCM Property implementing the "Remember Property" pattern -->
+    <Property Id="ENABLEGCM" Value="1">
+      <RegistrySearch Id="EnableGcmSearch" Root="HKLM" Key="SOFTWARE\GitForWindows" Name="EnableGcm" Type="raw" />
+    </Property>
+    <SetProperty Id="CMDLINE_ENABLEGCM" Value="[ENABLEGCM]" Before="AppSearch" Sequence="first" />
+    <SetProperty Id="ENABLEGCM" Value="[CMDLINE_ENABLEGCM]" After="AppSearch" Sequence="first">CMDLINE_ENABLEGCM</SetProperty>
+
     <Feature Id='Git'>
       <Component Directory='INSTALLFOLDER'>
-        <RegistryValue Root='HKLM' Key='Software\Git' Name='InstallFolder' Value='[INSTALLFOLDER]' Type='string' />
+        <RegistryValue Root='HKLM' Key='SOFTWARE\GitForWindows' Name='InstallFolder' Value='[INSTALLFOLDER]' Type='string' />
         <Environment Name='PATH' Value='[BinFolder]' Action='set' Part='last' System='yes' />
-
         <RemoveFolder Directory='GitProgramMenuFolder' On='uninstall' />
       </Component>
+      <Component Directory='ApplicationFolder'>
+        <RegistryValue Action='write' Type='string' Root='HKLM' Key='SOFTWARE\GitForWindows' Name='CliEnvironment' Value='[CLIENVIRONMENT]' />
+        <RemoveFolder Id='RemoveApplicationFolder' On='uninstall' />
+      </Component>
+      <Component Directory='ApplicationFolder'>
+        <RegistryValue Action='write' Type='string' Root='HKLM' Key='SOFTWARE\GitForWindows' Name='LineEndings' Value='[LINEENDINGS]' />
+      </Component>
+      <Component Directory='ApplicationFolder'>
+        <RegistryValue Action='write' Type='string' Root='HKLM' Key='SOFTWARE\GitForWindows' Name='Terminal' Value='[TERMINAL]' />
+      </Component>
+      <Component Directory='ApplicationFolder'>
+        <RegistryValue Action='write' Type='integer' Root='HKLM' Key='SOFTWARE\GitForWindows' Name='EnableFsCache' Value='[ENABLEFSCACHE]' />
+      </Component>
+      <Component Directory='ApplicationFolder'>
+        <RegistryValue Action='write' Type='integer' Root='HKLM' Key='SOFTWARE\GitForWindows' Name='EnableGcm' Value='[ENABLEGCM]' />
+      </Component>
```

Comment:
It would be really awesome if the `HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Git_is1` key could get the version as `DisplayVersion` value, as this is exactly the tell-tale used by, say, Visual Studio to determine whether Git for Windows is installed or not...